swift/logging/SwiftDiagnostics.h

```diff
@@ -46,34 +46,43 @@ struct SwiftDiagnostic {
     all = 0b111,
   };
 
+  // Notice that Tool Status Page severity is not necessarily the same as log severity, as the
+  // scope is different: TSP's scope is the whole analysis, log's scope is a single run
+  enum class Severity {
+    note,
+    warning,
+    error,
+  };
+
+  static constexpr std::string_view extractorName = "swift";
+
   std::string_view id;
   std::string_view name;
-  static constexpr std::string_view extractorName = "swift";
-  Format format;
   std::string_view action;
-  // space separated if more than 1. Not a vector to allow constexpr
-  // TODO(C++20) with vector going constexpr this can be turned to `std::vector<std::string_view>`
-  std::string_view helpLinks;
-  // for the moment, we only output errors, so no need to store the severity
 
+  Format format{Format::markdown};
   Visibility visibility{Visibility::all};
+  Severity severity{Severity::error};
+
+  // space separated if more than 1. Not a vector to allow constexpr
+  // TODO(C++20) with vector going constexpr this can be turned to `std::vector<std::string_view>`
+  std::string_view helpLinks{""};
 
   std::optional<SwiftDiagnosticsLocation> location{};
 
   // notice help links are really required only for plaintext messages, otherwise they should be
   // directly embedded in the markdown message
+  // optional arguments can be any of
+  // * std::string_view for setting helpLinks
```

Comment:
(I tried `static_assert` a bit, but did not get it to actually print the offending type, so the error was not very useful)

Reply:
> C++20 designated initializers
Sorry, completely forgot they were only added in C++20! (I've been using them in personal projects a lot longer; since they're also in C99, they've been a compiler extension in C++ for a long time.)

Reply:
yeah, I really wonder what took them so long to be 21 years behind C, of all languages :rofl: